internal/archive/archive.go

```diff
@@ -157,13 +194,30 @@ func openUbuntu(options *Options) (Archive, error) {
 	if len(options.Version) == 0 {
 		return nil, fmt.Errorf("archive options missing version")
 	}
+	if options.Pro != "" {
+		if _, ok := proArchiveInfo[options.Pro]; !ok {
+			return nil, fmt.Errorf("invalid pro value: %q", options.Pro)
+		}
+	}
+
+	baseURL := archiveURL(options.Pro, options.Arch)
```

Comment:
Logic above is suspect: we ensure the value is inside a map, to then call a function whose key purpose is to use the same map to return the value pointed to.  Move the error inside the function, and make it return that error, instead of verified the error case outside and then returning an invalid base URL if pro is wrong.

Reply:
Agreed, every time we have a pro archive we will need to get the URL and we can do the error validation there.

Reply:
Done in https://github.com/canonical/chisel/pull/167/commits/3d457a9f712677332fc7fec6d878962671d4ed3d.